src/semantics/bindingsInPattern.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ParseTreeVisitor} from '../syntax/ParseTreeVisitor.js';
+import {StringSet} from '../util/StringSet.js';
+
+class BindingsInPattern extends ParseTreeVisitor {
```

Comment:
FindBindingsInDestructuring?  Neither the class  nor the calllsite talk about patterns so I wasn't able to connect the dots. The callsite does talk about hasDestructuring().  So if it were hasDestructuringPattern(), then BindingsInPattern would be clear.